File: java-spring-rest-mvc/src/test/java/com/spring_guru/spring_rest_mvc/controllers/BeerControllerTest.java
package com.spring_guru.spring_rest_mvc.controllers;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.UUID;

@SpringBootTest
class BeerControllerTest {

    @Autowired
    BeerController controller;

    @Test
    void getBeerById() {
        System.out.println(controller.getBeerById(UUID.randomUUID()));
    }
}

File: kotlin-spring-rest-mvc/src/test/kotlin/com/kotlin_spring_rest_mvc/kotlin_spring_rest_mvc/controllers/BeerControllerTest.kt
package com.kotlin_spring_rest_mvc.kotlin_spring_rest_mvc.controllers

import com.kotlin_spring_rest_mvc.kotlin_spring_rest_mvc.services.BeerService
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
import java.util.*

@WebMvcTest(BeerController::class)
class BeerControllerTest {

    @Autowired
    lateinit var mockMvc: MockMvc

    @MockBean
    lateinit var service: BeerService

    @Test
    fun getBeerById() {
        mockMvc.perform(
            get("/api/v1/beers/" + UUID.randomUUID())
                .accept(MediaType.APPLICATION_JSON)
        )
            .andExpect(status().isOk)
    }
}